virttest/test_setup/__init__.py

```diff
@@ -515,22 +515,31 @@ def get_node_num_huge_pages(self, node, pagesize, type="total"):
         obj = kernel_interface.SysFS(node_page_path)
         return obj.sys_fs_value
 
-    def set_node_num_huge_pages(self, num, node, pagesize):
+    def set_node_num_huge_pages(self, num, node, pagesize, ignore_error=False):
         """
         Set number of pages of certain page size under given numa node.
 
         :param num: string or int, number of pages
         :param node: string or int, node number
         :param pagesize: string or int, page size in kB
+        :param ignore_error: True, not raise exception when failing
+        :return: boolean, False if page number is less than set value,
+                        otherwise, True
         """
         node_page_path = "%s/node%s" % (self.sys_node_path, node)
         node_page_path += "/hugepages/hugepages-%skB/nr_hugepages" % pagesize
         obj = kernel_interface.SysFS(node_page_path)
         obj.sys_fs_value = num
         # If node has some used hugepage, result will be larger than expected.
         if obj.sys_fs_value < int(num):
-            raise ValueError("Cannot set %s hugepages on node %s, please check"
-                             " if the node has enough memory" % (num, node))
+            if not ignore_error:
+                raise ValueError("Cannot set %s hugepages on node %s, please check"
+                                 " if the node has enough memory" % (num, node))
+            else:
+                LOG.warning("Only %s pages are allocated instead "
+                            "of %s pages", obj.sys_fs_value, int(num))
+                return False
+        return True
```

Comment:
Would you consider merging the error/waring message for these 2 scenarios?

Reply:
updated